framework/mesh/mesh_continuum/mesh_continuum.cc

```diff
@@ -954,13 +954,8 @@ MeshContinuum::CheckPointInsideCell(const Cell& cell, const Vector3& point) cons
     const auto& v0 = grid_ref.vertices[cell.vertex_ids_[0]];
     const auto& v1 = grid_ref.vertices[cell.vertex_ids_[1]];
 
-    const auto v01 = v1 - v0;
-    const auto v0p = point - v0;
-
-    const double v0p_dot_v01 = v0p.Dot(v01);
-
-    if (not(v0p_dot_v01 >= 0 and v0p_dot_v01 < v01.Norm()))
-      inside = false;
+    if (((v0.z - point.z) * (v1.z - point.z)) > 0.0)
+      return false;
```

Comment:
This routine is a quite messy... `inside` is set to `true` at first. For slabs, if a point is found to not be within a cell, `false` is returned within the `if` statement. For polygons, `inside` is set to `false` when that condition is encountered, and for polyhedra, `inside` is set to `false` at first and then a check of whether this a point is inside the cell is done to potentially change it to `true`. 

Can this be changed to be uniform across cell types? The variable `inside` doesn't seem to be necessary. For each cell type, check either if the cell is inside(outside) the cell (same for all cell types), return true(false) if the condition is met, and the have the last line of the routine be `return false(true)`?

Reply:
Whatever you would prefer is fine with me. I think we will all agree it should be rewritten, so the question is whether it gets rewritten now or later. If you feel like it isn't a big endeavor and want to rewrite it here, I'm not opposed. If not, an issue would be best before closing this PR.

Reply:
I rewrite it for consistency, and then open a PR to review the actual methods.

Reply:
Done.